dbcon/execplan/treenode.h

```diff
@@ -664,25 +665,19 @@ inline int64_t TreeNode::getIntVal()
   switch (fResultType.colDataType)
   {
     case CalpontSystemCatalog::CHAR:
-      if (fResultType.colWidth <= 8)
-        return fResult.intVal;
-
-      return atoll(fResult.strVal.c_str());
-
     case CalpontSystemCatalog::VARCHAR:
-      if (fResultType.colWidth <= 7)
-        return fResult.intVal;
-
-      return atoll(fResult.strVal.c_str());
-
-    // FIXME: ???
     case CalpontSystemCatalog::VARBINARY:
     case CalpontSystemCatalog::BLOB:
     case CalpontSystemCatalog::TEXT:
-      if (fResultType.colWidth <= 7)
-        return fResult.intVal;
+    {
+      // int expectedShortWidth = fResultType.colDataType == CalpontSystemCatalog::CHAR ? 8 : 7;
+      // if (fResultType.colWidth <= expectedShortWidth)
+      //    return fResult.intVal;
 
-      return atoll(fResult.strVal.c_str());
+      datatypes::DataCondition cnverr;
+      literal::Converter<literal::SignedInteger> cnv(fResult.strVal, cnverr);
+      return cnv.toXIntPositive<uint64_t>(cnverr);
```

Comment:
It is worth to log cnverr into stdout. This might reduce perf but we wll be at least aware about conversion errors.